vms/proposervm/block.go

```diff
@@ -141,13 +142,36 @@ func (p *postForkCommonComponents) Verify(
 			)
 		}
 
-		delay, err := p.verifyBlockDelay(ctx, parentTimestamp, parentPChainHeight, child)
-		if err != nil {
-			return err
+		shouldHaveProposer := true // post Durango this is always the case
+		if p.vm.IsDurangoActivated(parentTimestamp) {
+			err := p.verifyPostDurangoBlockDelay(ctx, parentTimestamp, parentPChainHeight, child)
+			if err != nil {
+				return err
+			}
+
+			// Post durango, p.vm.Ctx.NodeID slot may be so far in time that we don't
+			// schedule it when resetting vm preferred block. So we make sure to reschedule timer
+			// every time we verify a block to avoid stalls where innerVM is ready to build a block
+			// but proposerVM has not scheduled its timer.
+			nextStartTime, err := p.vm.resetPostDurangoScheduler(ctx, child, childPChainHeight)
+			switch err {
+			case nil:
+				p.vm.Scheduler.SetBuildBlockTime(nextStartTime)
+			case proposer.ErrNoSlotsScheduledInNextFuture:
+				// Can't schedule a slot now. We'll try later again
+			default:
+				return err
+			}
```

Comment:
I don't understand the goal of this piece of code.

1. We always build blocks on top of the preferred block, so we should never set the scheduler based on a potentially un-preferred block.
2. We only verify a block after one has been built... So I don't understand how this can guarantee that we will build a block.

Was this intended to be done in `BuildBlock` rather than `Verify`?

Reply:
Absolutelly yes, clearly I was allucinating